src/main/java/net/revelc/code/formatter/jsoup/JsoupBasedFormatter.java

```diff
@@ -71,36 +71,45 @@ public String doFormat(String code, final LineEnding ending) {
 
         var formattedCode = document.outerHtml();
 
-        // TODO: Fixing trailing space issue caused by jsoup. We do fix this during a proper run
+        // XXX: Fixing trailing space issue caused by jsoup. We do fix this during a proper run
         // but our tests fail to do so thus we are duplicating this until jsoup fixes bug.
         formattedCode = REMOVE_TRAILING_PATTERN.matcher(formattedCode).replaceAll("");
 
-        // TODO: Fixing jsoup counting issue which occurs when more than one indentation (jsoup can have mixed line
-        // ending content)
+        // XXX: jsoup processing results in mixed line ending content and needs normalized until jsoup fixes bug.
+        String[] lines = formattedCode.split("\\r?\\n");
+        formattedCode = String.join(ending.getChars(), lines);
+
+        // XXX: Fixing jsoup counter issue when more than one character indentation until jsoup fixes bug.
         if (this.formatter.indentAmount() > 1) {
-            int lineSize;
-            int newLineSize;
+            int lineLength;
+            int trimLineLength;
             int remainder;
-            String[] lines = formattedCode.split("\\r?\\n");
+            lines = formattedCode.split(ending.getChars());
```

Comment:
This line's change will leave behind extra chars if the detected line endings aren't consistent throughout the file (like if there's a mix of line ending styles in the file). The current code will split on any line, regardless of the line ending style.

Reply:
Added additional commits to help walk through changes on jsoup post pretty print processing.  Specific issue was done earlier in process to get it out of mixed patch so that each patch stands on its own.